src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -1822,6 +1816,18 @@ private int tryExecute(final Runnable runnable) {
         // not reached
     }
 
+    private int onSpinMisses(int spins) {
```

Comment:
I'm concerned that this might not be inlined.

Reply:
Let me print the ASM, but it should be due to the very small bytecode size

Reply:
That's it
```
....[Hottest Region 1]..............................................................................
c2, level 4, org.jboss.threads.EnhancedQueueExecutor::tryExecute, version 1403 (1045 bytes) 

                               0x00007f9003f3a724: shl    $0x3,%r10
                               0x00007f9003f3a728: movabs $0x800000000,%r12
                               0x00007f9003f3a732: add    %r12,%r10
                               0x00007f9003f3a735: xor    %r12,%r12
                               0x00007f9003f3a738: cmp    %r10,%rax
                               0x00007f9003f3a73b: jne    0x00007f8ffc316200  ;   {runtime_call ic_miss_stub}
                               0x00007f9003f3a741: data16 xchg %ax,%ax
                               0x00007f9003f3a744: nopl   0x0(%rax,%rax,1)
                               0x00007f9003f3a74c: data16 data16 xchg %ax,%ax
                             [Verified Entry Point]
  0.00%                        0x00007f9003f3a750: mov    %eax,-0x14000(%rsp)
  0.01%                        0x00007f9003f3a757: push   %rbp
  0.00%                        0x00007f9003f3a758: sub    $0x50,%rsp         ;*synchronization entry
                                                                             ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@-1 (line 1704)
  0.00%                        0x00007f9003f3a75c: mov    %rdx,%r10
  0.00%                        0x00007f9003f3a75f: mov    %rsi,(%rsp)
  0.00%                        0x00007f9003f3a763: mov    %rsi,%rdx
  0.00%                        0x00007f9003f3a766: add    $0x50,%rdx
                               0x00007f9003f3a76a: mov    (%rdx),%r8d        ;*getfield tail {reexecute=0 rethrow=0 return_oop=0}
                                                                             ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@11 (line 1705)
  0.05%                        0x00007f9003f3a76d: mov    %r10,%r13
  0.00%                        0x00007f9003f3a770: mov    %r10,%r11          ;*invokevirtual compareAndSetObject {reexecute=0 rethrow=0 return_oop=0}
                                                                             ; - sun.misc.Unsafe::compareAndSwapObject@9 (line 862)
                                                                             ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::compareAndSetTask@17 (line 2172)
                                                                             ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@133 (line 1745)
  0.00%                        0x00007f9003f3a773: mov    %r11d,0x10(%rsp)
                               0x00007f9003f3a778: xor    %r11d,%r11d
  0.00%                        0x00007f9003f3a77b: xor    %r10d,%r10d
  0.00%  ╭                     0x00007f9003f3a77e: jmp    0x00007f9003f3a79a
  0.01%  │  ↗                  0x00007f9003f3a780: pause                     ;*iload_1 {reexecute=0 rethrow=0 return_oop=0}
         │  │                                                                ; - org.jboss.threads.EnhancedQueueExecutor::onSpinMisses@34 (line 1828)
         │  │                                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@36 (line 1711)
  0.01%  │  │↗                 0x00007f9003f3a782: mov    (%rdx),%r8d        ;*goto {reexecute=0 rethrow=0 return_oop=0}
         │  ││                                                               ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@46 (line 1713)
  0.16%  │  ││                 0x00007f9003f3a785: mov    0x108(%r15),%r14   ; ImmutableOopMap{r10=Oop r8=NarrowOop rdx=Derived_oop_[0] r13=Oop [0]=Oop [16]=NarrowOop }
         │  ││                                                               ;*goto {reexecute=1 rethrow=0 return_oop=0}
         │  ││                                                               ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@46 (line 1713)
  0.00%  │  ││                 0x00007f9003f3a78c: test   %eax,(%r14)        ;*goto {reexecute=0 rethrow=0 return_oop=0}
         │  ││                                                               ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@46 (line 1713)
         │  ││                                                               ;   {poll}
  0.01%  │  ││                 0x00007f9003f3a78f: mov    (%rsp),%r9
  0.00%  │  ││      ↗       ↗  0x00007f9003f3a793: mov    %r9,(%rsp)
  0.00%  │  ││      │       │  0x00007f9003f3a797: mov    %esi,%r11d         ;*aload_3 {reexecute=0 rethrow=0 return_oop=0}
         │  ││      │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@21 (line 1709)
  0.00%  ↘  ││      │       │  0x00007f9003f3a79a: test   %r8d,%r8d
            ││      │       │  0x00007f9003f3a79d: je     0x00007f9003f3ac00  ;*invokevirtual getNext {reexecute=0 rethrow=0 return_oop=0}
            ││      │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@22 (line 1709)
  0.00%     ││      │       │  0x00007f9003f3a7a3: mov    %r11d,%esi
  0.00%     ││      │       │  0x00007f9003f3a7a6: inc    %esi               ;*iinc {reexecute=0 rethrow=0 return_oop=0}
            ││      │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::onSpinMisses@28 (line 1825)
            ││      │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@36 (line 1711)
  0.00%     ││      │       │  0x00007f9003f3a7a8: mov    %r8,%rdi           ;*aload_3 {reexecute=0 rethrow=0 return_oop=0}
            ││      │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@21 (line 1709)
  0.00%     ││      │       │  0x00007f9003f3a7ab: lea    0xc(%r8),%rbx
  0.00%     ││      │       │  0x00007f9003f3a7af: mov    (%rbx),%ebp        ;*getfield next {reexecute=0 rethrow=0 return_oop=0}
            ││      │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor$QNode::getNext@1 (line 2098)
            ││      │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@22 (line 1709)
  8.42%     ││      │       │  0x00007f9003f3a7b1: test   %ebp,%ebp
          ╭ ││      │       │  0x00007f9003f3a7b3: je     0x00007f9003f3a916
  0.03%   │ ││      │       │  0x00007f9003f3a7b9: mov    0x8(%rbp),%ecx
  1.22%   │ ││      │       │  0x00007f9003f3a7bc: cmp    $0x8012d01,%ecx    ;   {metadata(&apos;org/jboss/threads/EnhancedQueueExecutor$TaskNode&apos;)}
          │╭││      │       │  0x00007f9003f3a7c2: jne    0x00007f9003f3a7f1  ;*instanceof {reexecute=0 rethrow=0 return_oop=0}
          ││││      │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@27 (line 1710)
  0.01%   ││││      │       │  0x00007f9003f3a7c4: cmp    $0x80,%r11d
          ││╰│      │       │  0x00007f9003f3a7cb: jne    0x00007f9003f3a780  ;*if_icmpne {reexecute=0 rethrow=0 return_oop=0}
          ││ │      │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::onSpinMisses@17 (line 1821)
          ││ │      │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@36 (line 1711)
          ││ │      │       │  0x00007f9003f3a7cd: mov    %r10,0x18(%rsp)
  0.00%   ││ │      │       │  0x00007f9003f3a7d2: mov    %rdx,0x8(%rsp)
  0.00%   ││ │      │       │  0x00007f9003f3a7d7: mov    %r13,%rbp
          ││ │      │       │  0x00007f9003f3a7da: nop
          ││ │      │       │  0x00007f9003f3a7db: callq  0x00007f8ffc316980  ; ImmutableOopMap{rbp=Oop [0]=Oop [8]=Derived_oop_[0] [16]=NarrowOop [24]=Oop }
          ││ │      │       │                                                ;*invokestatic yield {reexecute=0 rethrow=0 return_oop=0}
          ││ │      │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::onSpinMisses@22 (line 1823)
          ││ │      │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@36 (line 1711)
          ││ │      │       │                                                ;   {static_call}
          ││ │      │       │  0x00007f9003f3a7e0: xor    %esi,%esi
          ││ │      │       │  0x00007f9003f3a7e2: mov    %rbp,%r13
          ││ │      │       │  0x00007f9003f3a7e5: mov    0x8(%rsp),%rdx
          ││ │      │       │  0x00007f9003f3a7ea: mov    0x18(%rsp),%r10
          ││ ╰      │       │  0x00007f9003f3a7ef: jmp    0x00007f9003f3a782
  0.01%   │↘        │       │  0x00007f9003f3a7f1: cmp    $0x8012c7a,%ecx    ;   {metadata(&apos;org/jboss/threads/EnhancedQueueExecutor$PoolThreadNode&apos;)}
          │         │       │  0x00007f9003f3a7f7: jne    0x00007f9003f3adc2  ;*instanceof {reexecute=0 rethrow=0 return_oop=0}
          │         │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@71 (line 1717)
  0.01%   │         │       │  0x00007f9003f3a7fd: mov    %rbp,%r9           ;*invokevirtual getNext {reexecute=0 rethrow=0 return_oop=0}
          │         │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@78 (line 1718)
  0.00%   │         │       │  0x00007f9003f3a800: mov    0xc(%r9),%ebp      ;*getfield next {reexecute=0 rethrow=0 return_oop=0}
          │         │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor$QNode::getNext@1 (line 2098)
          │         │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::getNext@1 (line 2180)
          │         │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::getNext@1 (line 2124)
          │         │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@78 (line 1718)
  0.01%   │         │       │  0x00007f9003f3a804: test   %ebp,%ebp
          │         │       │  0x00007f9003f3a806: je     0x00007f9003f3abca
  0.01%   │         │       │  0x00007f9003f3a80c: mov    0x8(%rbp),%ecx
  0.21%   │         │       │  0x00007f9003f3a80f: cmp    $0x8012c7a,%ecx    ;   {metadata(&apos;org/jboss/threads/EnhancedQueueExecutor$PoolThreadNode&apos;)}
          │         │       │  0x00007f9003f3a815: jne    0x00007f9003f3ade2
  0.00%   │         │       │  0x00007f9003f3a81b: mov    %rbp,%r8
  0.01%   │         │       │  0x00007f9003f3a81e: mov    %r8,%rcx
  0.00%   │         │       │  0x00007f9003f3a821: mov    %r9,%rax
  0.00%   │         │       │  0x00007f9003f3a824: lock cmpxchg %ecx,(%rbx)
  9.04%   │         │       │  0x00007f9003f3a828: sete   %cl
  0.01%   │         │       │  0x00007f9003f3a82b: movzbl %cl,%ecx
  0.00%   │         │       │  0x00007f9003f3a82e: test   %ecx,%ecx
          │   ╭     │       │  0x00007f9003f3a830: je     0x00007f9003f3a848
  0.02%   │   │     │       │  0x00007f9003f3a832: mov    %rbx,%r8
  0.00%   │   │     │       │  0x00007f9003f3a835: shr    $0x9,%r8
  0.01%   │   │     │       │  0x00007f9003f3a839: movabs $0x7f8ffb20e000,%rbx
  0.00%   │   │     │       │  0x00007f9003f3a843: movb   $0x0,(%rbx,%r8,1)  ;*invokevirtual compareAndSetObject {reexecute=0 rethrow=0 return_oop=0}
          │   │     │       │                                                ; - sun.misc.Unsafe::compareAndSwapObject@9 (line 862)
          │   │     │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor$QNode::compareAndSetNext@9 (line 2094)
          │   │     │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@87 (line 1733)
  0.03%   │   ↘     │       │  0x00007f9003f3a848: test   %ecx,%ecx
          │    ╭    │       │  0x00007f9003f3a84a: je     0x00007f9003f3a8d1  ;*ifeq {reexecute=0 rethrow=0 return_oop=0}
          │    │    │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@90 (line 1733)
  0.00%   │    │    │       │  0x00007f9003f3a850: mov    %r9,%r8
  0.00%   │    │    │       │  0x00007f9003f3a853: add    $0x58,%r8
  0.00%   │    │    │       │  0x00007f9003f3a857: mov    (%r8),%ebp         ;*getfield task {reexecute=0 rethrow=0 return_oop=0}
          │    │    │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::compareAndSetTask@1 (line 2172)
          │    │    │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@133 (line 1745)
  0.23%   │    │    │       │  0x00007f9003f3a85a: cmp    $0xf571d460,%ebp   ;   {oop(a &apos;org/jboss/threads/NullRunnable&apos;{0x00000000f571d460})}
          │    │    │       │  0x00007f9003f3a860: jne    0x00007f9003f3ac8d
  0.00%   │    │    │       │  0x00007f9003f3a866: mov    $0xf571d460,%eax   ;   {oop(a &apos;org/jboss/threads/NullRunnable&apos;{0x00000000f571d460})}
  0.00%   │    │    │       │  0x00007f9003f3a86b: mov    0x10(%rsp),%ecx
  0.00%   │    │    │       │  0x00007f9003f3a86f: lock cmpxchg %ecx,(%r8)
  0.42%   │    │    │       │  0x00007f9003f3a874: sete   %bpl
  0.00%   │    │    │       │  0x00007f9003f3a878: movzbl %bpl,%ebp
  0.00%   │    │    │       │  0x00007f9003f3a87c: test   %ebp,%ebp
          │    │╭   │       │  0x00007f9003f3a87e: je     0x00007f9003f3a893
  0.00%   │    ││   │       │  0x00007f9003f3a880: shr    $0x9,%r8
          │    ││   │       │  0x00007f9003f3a884: movabs $0x7f8ffb20e000,%rcx
          │    ││   │       │  0x00007f9003f3a88e: movb   $0x0,(%rcx,%r8,1)  ;*invokevirtual compareAndSetObject {reexecute=0 rethrow=0 return_oop=0}
          │    ││   │       │                                                ; - sun.misc.Unsafe::compareAndSwapObject@9 (line 862)
          │    ││   │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::compareAndSetTask@17 (line 2172)
          │    ││   │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@133 (line 1745)
  0.00%   │    │↘   │       │  0x00007f9003f3a893: test   %ebp,%ebp
          │    │    │       │  0x00007f9003f3a895: je     0x00007f9003f3acae  ;*ifeq {reexecute=0 rethrow=0 return_oop=0}
          │    │    │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::compareAndSetTask@20 (line 2172)
          │    │    │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@133 (line 1745)
  0.00%   │    │    │       │  0x00007f9003f3a89b: mov    0x50(%r9),%r11d    ;*getfield parked {reexecute=0 rethrow=0 return_oop=0}
          │    │    │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::unpark@1 (line 2250)
          │    │    │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@151 (line 1747)
  0.08%   │    │    │       │  0x00007f9003f3a89f: test   %r11d,%r11d
          │    │ ╭  │       │  0x00007f9003f3a8a2: jne    0x00007f9003f3a8f6
  0.00%   │    │ │  │       │  0x00007f9003f3a8a4: mov    $0x2,%r10d
  0.00%   │    │ │  │       │  0x00007f9003f3a8aa: xor    %eax,%eax
  0.00%   │    │ │  │       │  0x00007f9003f3a8ac: lock cmpxchg %r10d,0x50(%r9)
  0.36%   │    │ │  │       │  0x00007f9003f3a8b2: sete   %r11b
  0.00%   │    │ │  │       │  0x00007f9003f3a8b6: movzbl %r11b,%r11d        ;*invokevirtual compareAndSetInt {reexecute=0 rethrow=0 return_oop=0}
          │    │ │  │       │                                                ; - sun.misc.Unsafe::compareAndSwapInt@9 (line 878)
          │    │ │  │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::unpark@16 (line 2250)
          │    │ │  │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@151 (line 1747)
  0.00%   │    │ │  │       │  0x00007f9003f3a8ba: test   %r11d,%r11d
          │    │ │╭ │       │  0x00007f9003f3a8bd: je     0x00007f9003f3a8f6  ;*ifeq {reexecute=0 rethrow=0 return_oop=0}
          │    │ ││ │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::isShutdownRequested@7 (line 1997)
          │    │ ││ │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryAllocateThread@6 (line 1584)
          │    │ ││ │       │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@181 (line 1757)
  0.00%   │    │ ││ │↗      │  0x00007f9003f3a8bf: xor    %eax,%eax
          │    │ ││ ││      │  0x00007f9003f3a8c1: add    $0x50,%rsp
  0.00%   │    │ ││ ││      │  0x00007f9003f3a8c5: pop    %rbp
  0.00%   │    │ ││ ││      │  0x00007f9003f3a8c6: mov    0x108(%r15),%r10
  0.00%   │    │ ││ ││      │  0x00007f9003f3a8cd: test   %eax,(%r10)        ;   {poll_return}
  0.00%   │    │ ││ ││      │  0x00007f9003f3a8d0: retq   
  0.01%   │    ↘ ││ ││      │  0x00007f9003f3a8d1: cmp    $0x80,%r11d
          │      ││╭││      │  0x00007f9003f3a8d8: je     0x00007f9003f3ab6a  ;*if_icmpne {reexecute=0 rethrow=0 return_oop=0}
          │      │││││      │                                                ; - org.jboss.threads.EnhancedQueueExecutor::onSpinMisses@17 (line 1821)
          │      │││││      │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@159 (line 1752)
  0.00%   │      │││││      │  0x00007f9003f3a8de: pause                     ;*iload_1 {reexecute=0 rethrow=0 return_oop=0}
          │      │││││      │                                                ; - org.jboss.threads.EnhancedQueueExecutor::onSpinMisses@34 (line 1828)
          │      │││││      │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@159 (line 1752)
  0.03%   │      │││││      │  0x00007f9003f3a8e0: mov    (%rdx),%r8d        ;*goto {reexecute=0 rethrow=0 return_oop=0}
          │      │││││      │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@46 (line 1713)
  0.12%   │      │││││      │  0x00007f9003f3a8e3: mov    0x108(%r15),%r14   ; ImmutableOopMap{r10=Oop r8=NarrowOop rdx=Derived_oop_[0] r13=Oop [0]=Oop [16]=NarrowOop }
          │      │││││      │                                                ;*goto {reexecute=1 rethrow=0 return_oop=0}
          │      │││││      │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@169 (line 1755)
  0.00%   │      │││││      │  0x00007f9003f3a8ea: test   %eax,(%r14)        ;*goto {reexecute=0 rethrow=0 return_oop=0}
          │      │││││      │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@169 (line 1755)
          │      │││││      │                                                ;   {poll}
  0.02%   │      │││││      │  0x00007f9003f3a8ed: mov    (%rsp),%r9
  0.00%   │      │││╰│      │  0x00007f9003f3a8f1: jmpq   0x00007f9003f3a793  ;*ifeq {reexecute=0 rethrow=0 return_oop=0}
          │      │││ │      │                                                ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::unpark@19 (line 2250)
          │      │││ │      │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@151 (line 1747)
  0.00%   │      ↘↘│ │      │  0x00007f9003f3a8f6: mov    0x54(%r9),%ebp     ;*getfield thread {reexecute=0 rethrow=0 return_oop=0}
          │        │ │      │                                                ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::unpark@24 (line 2253)
          │        │ │      │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@151 (line 1747)
  0.00%   │        │ │      │  0x00007f9003f3a8fa: test   %ebp,%ebp
          │        │ │      │  0x00007f9003f3a8fc: je     0x00007f9003f3acce  ;*ifnull {reexecute=0 rethrow=0 return_oop=0}
          │        │ │      │                                                ; - java.util.concurrent.locks.LockSupport::unpark@1 (line 159)
          │        │ │      │                                                ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::unpark@27 (line 2253)
          │        │ │      │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@151 (line 1747)
  0.00%   │        │ │      │  0x00007f9003f3a902: mov    %rbp,%rdx          ;*getfield thread {reexecute=0 rethrow=0 return_oop=0}
          │        │ │      │                                                ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::unpark@24 (line 2253)
          │        │ │      │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@151 (line 1747)
  0.00%   │        │ │      │  0x00007f9003f3a905: movabs $0xf56100c8,%rsi   ;   {oop(a &apos;jdk/internal/misc/Unsafe&apos;{0x00000000f56100c8})}
  0.00%   │        │ │      │  0x00007f9003f3a90f: callq  0x00007f8ffc316480  ; ImmutableOopMap{}
          │        │ │      │                                                ;*invokevirtual unpark {reexecute=0 rethrow=0 return_oop=0}
          │        │ │      │                                                ; - java.util.concurrent.locks.LockSupport::unpark@8 (line 160)
          │        │ │      │                                                ; - org.jboss.threads.EnhancedQueueExecutor$PoolThreadNode::unpark@27 (line 2253)
          │        │ │      │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@151 (line 1747)
          │        │ │      │                                                ;   {optimized virtual_call}
  0.00%   │        │ ╰      │  0x00007f9003f3a914: jmp    0x00007f9003f3a8bf
  0.04%   ↘        │        │  0x00007f9003f3a916: mov    (%rsp),%r9
  0.02%            │        │  0x00007f9003f3a91a: vmovss 0x130(%r9),%xmm1   ;*getfield growthResistance {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@178 (line 1757)
  0.34%            │        │  0x00007f9003f3a923: mov    0xd8(%r9),%r14     ;*getfield threadStatus {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryAllocateThread@1 (line 1583)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@181 (line 1757)
  0.08%            │        │  0x00007f9003f3a92a: movabs $0x2000000000000000,%rcx
  0.00%            │        │  0x00007f9003f3a934: mov    %r14,%rax
                   │        │  0x00007f9003f3a937: and    %rcx,%rax          ;*land {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::isShutdownRequested@4 (line 1997)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryAllocateThread@6 (line 1584)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@181 (line 1757)
  0.00%            │        │  0x00007f9003f3a93a: test   %rax,%rax
                   │        │  0x00007f9003f3a93d: jne    0x00007f9003f3ad3a  ;*ifeq {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::isShutdownRequested@7 (line 1997)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryAllocateThread@6 (line 1584)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@181 (line 1757)
  0.00%            │        │  0x00007f9003f3a943: mov    %r14,%rcx
  0.00%            │        │  0x00007f9003f3a946: shr    $0x28,%rcx
  0.00%            │        │  0x00007f9003f3a94a: and    $0xfffff,%rcx
  0.00%            │        │  0x00007f9003f3a951: mov    %ecx,%ecx          ;*l2i {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::maxSizeOf@8 (line 1958)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryAllocateThread@21 (line 1588)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@181 (line 1757)
  0.00%            │        │  0x00007f9003f3a953: mov    %r14,%rax
  0.00%            │        │  0x00007f9003f3a956: and    $0xfffff,%rax
  0.00%            │        │  0x00007f9003f3a95d: mov    %eax,%ebp          ;*l2i {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::currentSizeOf@7 (line 1962)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryAllocateThread@15 (line 1587)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@181 (line 1757)
  0.00%            │        │  0x00007f9003f3a95f: cmp    %ecx,%ebp
                   │        │  0x00007f9003f3a961: jl     0x00007f9003f3ad0a  ;*if_icmplt {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryAllocateThread@24 (line 1588)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@181 (line 1757)
  0.00%            │        │  0x00007f9003f3a967: mov    0x120(%r9),%rax    ;*getfield queueSize {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@1 (line 1893)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  1.47%            │        │  0x00007f9003f3a96e: mov    %rax,%rcx
  0.00%            │        │  0x00007f9003f3a971: and    $0x7fffffff,%rcx
  0.00%            │        │  0x00007f9003f3a978: mov    %ecx,%ecx          ;*l2i {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::currentQueueSizeOf@5 (line 1936)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@6 (line 1894)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  0.00%            │        │  0x00007f9003f3a97a: mov    %rax,%rbp
  0.00%            │        │  0x00007f9003f3a97d: shr    $0x20,%rbp
  0.00%            │        │  0x00007f9003f3a981: and    $0x7fffffff,%rbp
  0.00%            │        │  0x00007f9003f3a988: mov    %ebp,%ebp          ;*l2i {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::maxQueueSizeOf@8 (line 1945)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@12 (line 1895)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  0.00%            │        │  0x00007f9003f3a98a: cmp    %ebp,%ecx
                   │        │  0x00007f9003f3a98c: jge    0x00007f9003f3ad7a
  0.00%            │        │  0x00007f9003f3a992: inc    %ecx
  0.00%            │        │  0x00007f9003f3a994: movslq %ecx,%r14
  0.00%            │        │  0x00007f9003f3a997: mov    %rax,%rbp
  0.00%            │        │  0x00007f9003f3a99a: movabs $0xffffffff00000000,%rcx
  0.00%            │        │  0x00007f9003f3a9a4: and    %rcx,%rbp
  0.00%            │        │  0x00007f9003f3a9a7: or     %r14,%rbp
                   │        │  0x00007f9003f3a9aa: lock cmpxchg %rbp,0x120(%r9)
  3.24%            │        │  0x00007f9003f3a9b3: sete   %cl
  0.00%            │        │  0x00007f9003f3a9b6: movzbl %cl,%ecx           ;*invokevirtual compareAndSetLong {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - sun.misc.Unsafe::compareAndSwapLong@9 (line 894)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::compareAndSetQueueSize@9 (line 1877)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@33 (line 1900)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  0.00%            │        │  0x00007f9003f3a9b9: test   %ecx,%ecx
                   │  ╭     │  0x00007f9003f3a9bb: jne    0x00007f9003f3aa20  ;*ifne {reexecute=0 rethrow=0 return_oop=0}
                   │  │     │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@36 (line 1900)
                   │  │     │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  0.01%            │  │     │  0x00007f9003f3a9bd: data16 xchg %ax,%ax       ;*getstatic UPDATE_STATISTICS {reexecute=0 rethrow=0 return_oop=0}
                   │  │     │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@39 (line 1901)
                   │  │     │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  0.01%            │  │↗    │  0x00007f9003f3a9c0: mov    0x120(%r9),%rax    ;*getfield queueSize {reexecute=0 rethrow=0 return_oop=0}
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@53 (line 1902)
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  0.22%            │  ││    │  0x00007f9003f3a9c7: mov    %rax,%rcx
  0.00%            │  ││    │  0x00007f9003f3a9ca: and    $0x7fffffff,%rcx
  0.00%            │  ││    │  0x00007f9003f3a9d1: mov    %ecx,%ecx          ;*l2i {reexecute=0 rethrow=0 return_oop=0}
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::currentQueueSizeOf@5 (line 1936)
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@58 (line 1903)
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  0.00%            │  ││    │  0x00007f9003f3a9d3: mov    %rax,%rbp
  0.00%            │  ││    │  0x00007f9003f3a9d6: shr    $0x20,%rbp
                   │  ││    │  0x00007f9003f3a9da: and    $0x7fffffff,%rbp
  0.00%            │  ││    │  0x00007f9003f3a9e1: mov    %ebp,%ebp          ;*l2i {reexecute=0 rethrow=0 return_oop=0}
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::maxQueueSizeOf@8 (line 1945)
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@64 (line 1904)
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  0.00%            │  ││    │  0x00007f9003f3a9e3: cmp    %ebp,%ecx
                   │  ││    │  0x00007f9003f3a9e5: jge    0x00007f9003f3acda  ;*if_icmplt {reexecute=0 rethrow=0 return_oop=0}
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@67 (line 1904)
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  0.00%            │  ││    │  0x00007f9003f3a9eb: inc    %ecx               ;*goto {reexecute=0 rethrow=0 return_oop=0}
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@46 (line 1713)
                   │  ││    │  0x00007f9003f3a9ed: mov    0x108(%r15),%r14   ; ImmutableOopMap{r10=Oop r8=NarrowOop r9=Oop rbx=Derived_oop_rdi rdi=Oop rdx=Derived_oop_r9 r13=Oop [16]=NarrowOop }
                   │  ││    │                                                ;*goto {reexecute=1 rethrow=0 return_oop=0}
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@77 (line 1908)
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  0.00%            │  ││    │  0x00007f9003f3a9f4: test   %eax,(%r14)        ;   {poll}
  0.00%            │  ││    │  0x00007f9003f3a9f7: movslq %ecx,%rcx
  0.00%            │  ││    │  0x00007f9003f3a9fa: mov    %rax,%rbp
                   │  ││    │  0x00007f9003f3a9fd: movabs $0xffffffff00000000,%r14
                   │  ││    │  0x00007f9003f3aa07: and    %r14,%rbp
  0.00%            │  ││    │  0x00007f9003f3aa0a: or     %rcx,%rbp
  0.00%            │  ││    │  0x00007f9003f3aa0d: lock cmpxchg %rbp,0x120(%r9)
  1.79%            │  ││    │  0x00007f9003f3aa16: sete   %cl
  0.00%            │  ││    │  0x00007f9003f3aa19: movzbl %cl,%ecx           ;*invokevirtual compareAndSetLong {reexecute=0 rethrow=0 return_oop=0}
                   │  ││    │                                                ; - sun.misc.Unsafe::compareAndSwapLong@9 (line 894)
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::compareAndSetQueueSize@9 (line 1877)
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@33 (line 1900)
                   │  ││    │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  0.00%            │  ││    │  0x00007f9003f3aa1c: test   %ecx,%ecx
                   │  │╰    │  0x00007f9003f3aa1e: je     0x00007f9003f3a9c0  ;*ifne {reexecute=0 rethrow=0 return_oop=0}
                   │  │     │                                                ; - org.jboss.threads.EnhancedQueueExecutor::increaseQueueSize@36 (line 1900)
                   │  │     │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@248 (line 1768)
  0.01%            │  ↘     │  0x00007f9003f3aa20: test   %r10,%r10
                   │    ╭   │  0x00007f9003f3aa23: jne    0x00007f9003f3aa90
  0.00%            │    │   │  0x00007f9003f3aa25: mov    0x118(%r15),%r10
  0.00%            │    │   │  0x00007f9003f3aa2c: mov    %r10,%rcx
  0.00%            │    │   │  0x00007f9003f3aa2f: add    $0x18,%rcx
  0.00%            │    │   │  0x00007f9003f3aa33: cmp    0x128(%r15),%rcx
                   │    │   │  0x00007f9003f3aa3a: jae    0x00007f9003f3ac0e  ;*goto {reexecute=0 rethrow=0 return_oop=0}
                   │    │   │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@46 (line 1713)
  0.00%            │    │   │  0x00007f9003f3aa40: mov    %rcx,0x118(%r15)
  0.00%            │    │   │  0x00007f9003f3aa47: prefetchw 0xc0(%rcx)
  0.00%            │    │   │  0x00007f9003f3aa4e: movq   $0x1,(%r10)
  0.00%            │    │   │  0x00007f9003f3aa55: movl   $0x8012d01,0x8(%r10)  ;   {metadata(&apos;org/jboss/threads/EnhancedQueueExecutor$TaskNode&apos;)}
  0.00%            │    │   │  0x00007f9003f3aa5d: movl   $0x0,0xc(%r10)
  0.00%            │    │   │  0x00007f9003f3aa65: movq   $0x0,0x10(%r10)
  0.00%            │    │   │  0x00007f9003f3aa6d: mov    0x10(%rsp),%ecx
  0.00%            │    │   │  0x00007f9003f3aa71: mov    %ecx,0x10(%r10)
  0.00%            │    │   │  0x00007f9003f3aa75: mov    %r10,%rcx
  0.00%            │    │   │  0x00007f9003f3aa78: shr    $0x9,%rcx
  0.00%            │    │   │  0x00007f9003f3aa7c: movabs $0x7f8ffb20e000,%rdi
  0.00%            │    │   │  0x00007f9003f3aa86: movb   $0x0,(%rdi,%rcx,1)
  0.00%            │    │   │  0x00007f9003f3aa8a: lock addl $0x0,-0x40(%rsp)
  0.79%            │    ↘   │  0x00007f9003f3aa90: mov    %r10,%rdi
  0.00%            │        │  0x00007f9003f3aa93: xor    %rax,%rax
  0.00%            │        │  0x00007f9003f3aa96: lock cmpxchg %edi,(%rbx)
  2.49%            │        │  0x00007f9003f3aa9a: sete   %cl
  0.00%            │        │  0x00007f9003f3aa9d: movzbl %cl,%ecx
  0.00%            │        │  0x00007f9003f3aaa0: test   %ecx,%ecx
                   │     ╭  │  0x00007f9003f3aaa2: je     0x00007f9003f3aab6
  0.00%            │     │  │  0x00007f9003f3aaa4: shr    $0x9,%rbx
  0.00%            │     │  │  0x00007f9003f3aaa8: movabs $0x7f8ffb20e000,%rax
                   │     │  │  0x00007f9003f3aab2: movb   $0x0,(%rax,%rbx,1)  ;*invokevirtual compareAndSetObject {reexecute=0 rethrow=0 return_oop=0}
                   │     │  │                                                ; - sun.misc.Unsafe::compareAndSwapObject@9 (line 862)
                   │     │  │                                                ; - org.jboss.threads.EnhancedQueueExecutor$QNode::compareAndSetNext@9 (line 2094)
                   │     │  │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@327 (line 1796)
  0.01%            │     ↘  │  0x00007f9003f3aab6: test   %ecx,%ecx
                   │        │  0x00007f9003f3aab8: jne    0x00007f9003f3ab90  ;*ifeq {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@330 (line 1796)
  0.00%            │        │  0x00007f9003f3aabe: mov    0x120(%r9),%rax
  1.57%            │        │  0x00007f9003f3aac5: mov    %rax,%rbx
  0.00%            │        │  0x00007f9003f3aac8: movabs $0xffffffff00000000,%r8
                   │        │  0x00007f9003f3aad2: and    %r8,%rbx
  0.00%            │        │  0x00007f9003f3aad5: mov    %rax,%r8
  0.00%            │        │  0x00007f9003f3aad8: and    $0x7fffffff,%r8
  0.00%            │        │  0x00007f9003f3aadf: mov    %r8d,%ecx
                   │        │  0x00007f9003f3aae2: dec    %ecx
  0.00%            │        │  0x00007f9003f3aae4: movslq %ecx,%r8
  0.00%            │        │  0x00007f9003f3aae7: or     %r8,%rbx
  0.00%            │        │  0x00007f9003f3aaea: lock cmpxchg %rbx,0x120(%r9)
  1.06%            │        │  0x00007f9003f3aaf3: sete   %r8b
  0.00%            │        │  0x00007f9003f3aaf7: movzbl %r8b,%r8d          ;*invokevirtual compareAndSetLong {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - sun.misc.Unsafe::compareAndSwapLong@9 (line 894)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::compareAndSetQueueSize@9 (line 1877)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::decreaseQueueSize@38 (line 1924)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@359 (line 1804)
  0.00%            │        │  0x00007f9003f3aafb: test   %r8d,%r8d
                   │      ╭ │  0x00007f9003f3aafe: jne    0x00007f9003f3ab4d  ;*getstatic UPDATE_STATISTICS {reexecute=0 rethrow=0 return_oop=0}
                   │      │ │                                                ; - org.jboss.threads.EnhancedQueueExecutor::decreaseQueueSize@44 (line 1925)
                   │      │ │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@359 (line 1804)
  0.01%            │      │↗│  0x00007f9003f3ab00: mov    0x120(%r9),%rax    ;*getfield queueSize {reexecute=0 rethrow=0 return_oop=0}
                   │      │││                                                ; - org.jboss.threads.EnhancedQueueExecutor::decreaseQueueSize@58 (line 1926)
                   │      │││                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@359 (line 1804)
  0.14%            │      │││  0x00007f9003f3ab07: mov    %rax,%rbx
  0.00%            │      │││  0x00007f9003f3ab0a: movabs $0xffffffff00000000,%r8
                   │      │││  0x00007f9003f3ab14: and    %r8,%rbx           ;*goto {reexecute=0 rethrow=0 return_oop=0}
                   │      │││                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@46 (line 1713)
  0.00%            │      │││  0x00007f9003f3ab17: mov    0x108(%r15),%r14   ; ImmutableOopMap{r10=Oop r9=Oop rdx=Derived_oop_r9 r13=Oop [16]=NarrowOop }
                   │      │││                                                ;*ifne {reexecute=1 rethrow=0 return_oop=0}
                   │      │││                                                ; - org.jboss.threads.EnhancedQueueExecutor::decreaseQueueSize@65 (line 1927)
                   │      │││                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@359 (line 1804)
  0.00%            │      │││  0x00007f9003f3ab1e: test   %eax,(%r14)        ;   {poll}
  0.00%            │      │││  0x00007f9003f3ab21: mov    %rax,%r8
                   │      │││  0x00007f9003f3ab24: and    $0x7fffffff,%r8
  0.00%            │      │││  0x00007f9003f3ab2b: mov    %r8d,%r8d
  0.00%            │      │││  0x00007f9003f3ab2e: dec    %r8d
  0.00%            │      │││  0x00007f9003f3ab31: movslq %r8d,%r8
  0.00%            │      │││  0x00007f9003f3ab34: or     %r8,%rbx
  0.00%            │      │││  0x00007f9003f3ab37: lock cmpxchg %rbx,0x120(%r9)
  0.43%            │      │││  0x00007f9003f3ab40: sete   %r8b
  0.00%            │      │││  0x00007f9003f3ab44: movzbl %r8b,%r8d          ;*invokevirtual compareAndSetLong {reexecute=0 rethrow=0 return_oop=0}
                   │      │││                                                ; - sun.misc.Unsafe::compareAndSwapLong@9 (line 894)
                   │      │││                                                ; - org.jboss.threads.EnhancedQueueExecutor::compareAndSetQueueSize@9 (line 1877)
                   │      │││                                                ; - org.jboss.threads.EnhancedQueueExecutor::decreaseQueueSize@38 (line 1924)
                   │      │││                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@359 (line 1804)
  0.00%            │      │││  0x00007f9003f3ab48: test   %r8d,%r8d
                   │      │╰│  0x00007f9003f3ab4b: je     0x00007f9003f3ab00  ;*ifne {reexecute=0 rethrow=0 return_oop=0}
                   │      │ │                                                ; - org.jboss.threads.EnhancedQueueExecutor::decreaseQueueSize@41 (line 1924)
                   │      │ │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@359 (line 1804)
  0.00%            │      ↘ │  0x00007f9003f3ab4d: cmp    $0x80,%r11d
                   │        │  0x00007f9003f3ab54: je     0x00007f9003f3abd2  ;*if_icmpne {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::onSpinMisses@17 (line 1821)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@365 (line 1805)
  0.00%            │        │  0x00007f9003f3ab56: pause                     ;*iload_1 {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::onSpinMisses@34 (line 1828)
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@365 (line 1805)
  0.01%            │        │  0x00007f9003f3ab58: mov    (%rdx),%r8d        ;*goto {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@46 (line 1713)
  0.14%            │        │  0x00007f9003f3ab5b: mov    0x108(%r15),%r14   ; ImmutableOopMap{r10=Oop r8=NarrowOop r9=Oop rdx=Derived_oop_r9 r13=Oop [16]=NarrowOop }
                   │        │                                                ;*goto {reexecute=1 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@375 (line 1808)
  0.00%            │        │  0x00007f9003f3ab62: test   %eax,(%r14)        ;*goto {reexecute=0 rethrow=0 return_oop=0}
                   │        │                                                ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@375 (line 1808)
                   │        │                                                ;   {poll}
  0.00%            │        ╰  0x00007f9003f3ab65: jmpq   0x00007f9003f3a793
                   ↘           0x00007f9003f3ab6a: mov    %r10,0x18(%rsp)
                               0x00007f9003f3ab6f: mov    %rdx,0x8(%rsp)
                               0x00007f9003f3ab74: mov    %r13,%rbp
                               0x00007f9003f3ab77: callq  0x00007f8ffc316980  ; ImmutableOopMap{rbp=Oop [0]=Oop [8]=Derived_oop_[0] [16]=NarrowOop [24]=Oop }
                                                                             ;*invokestatic yield {reexecute=0 rethrow=0 return_oop=0}
                                                                             ; - org.jboss.threads.EnhancedQueueExecutor::onSpinMisses@22 (line 1823)
                                                                             ; - org.jboss.threads.EnhancedQueueExecutor::tryExecute@159 (line 1752)
                                                                             ;   {static_call}
....................................................................................................
 34.61%  <total for region 1>
```
`onSpinMisses` is always inlined

Reply:
Without a byte to spare! MaxInlineSize default value  is 35 bytes.
```
private int onSpinMisses(int)
     0: getstatic       #67  // Field UPDATE_STATISTICS:Z
     3: ifeq            13   
     6: aload_0         
     7: getfield        #3   // Field spinMisses:Ljava/util/concurrent/atomic/LongAdder;
    10: invokevirtual   #68  // Method java/util/concurrent/atomic/LongAdder.increment:()V
    13: iload_1         
    14: getstatic       #190 // Field YIELD_SPINS:I
    17: if_icmpne       28   
    20: iconst_0        
    21: istore_1        
    22: invokestatic    #191 // Method java/lang/Thread.yield:()V
    25: goto            34   
    28: iinc            1, 1 
    31: invokestatic    #192 // Method org/jboss/threads/JDKSpecific.onSpinWait:()V
    34: iload_1         
    35: ireturn
```

We could simplify a bit to

```java
    private int onSpinMisses(int spins) {
        if (UPDATE_STATISTICS) spinMisses.increment();
        if (spins == YIELD_SPINS) {
            Thread.yield();
            return 0;
        } else {
            JDKSpecific.onSpinWait();
            return spins + 1;
        }
    }
```

which produces:

```
private int onSpinMisses(int)
     0: getstatic       #67  // Field UPDATE_STATISTICS:Z
     3: ifeq            13   
     6: aload_0         
     7: getfield        #3   // Field spinMisses:Ljava/util/concurrent/atomic/LongAdder;
    10: invokevirtual   #68  // Method java/util/concurrent/atomic/LongAdder.increment:()V
    13: iload_1         
    14: getstatic       #190 // Field YIELD_SPINS:I
    17: if_icmpne       25   
    20: invokestatic    #191 // Method java/lang/Thread.yield:()V
    23: iconst_0        
    24: ireturn         
    25: invokestatic    #192 // Method org/jboss/threads/JDKSpecific.onSpinWait:()V
    28: iload_1         
    29: iconst_1        
    30: iadd            
    31: ireturn 
```

Reply:
I like the idea to leave a bit more space and probably worth put a comment on the method as well !